.travis.yml

```diff
@@ -5,35 +5,47 @@ env:
     # an encrypted 'GITHUB_TOKEN=<token>' for the PolymerLabs/arcs repo
     secure: "qFfvf0YAK5NWkH5fj708Z+1qIHtOFrmNhslB+n14rt5J3fJjY0A/4lbLbhx2jACYME+OU2m66maxh5abEZCF4ZGi4zpJa01IUYesTTa3qdM9kozB2TfFcl0d1xffuGpAh9kPchaadCvx/BU9fYFzkd8CLD5mQWCybwqM+6xpoO/KULDXhlcOPGbzg65xmPJANZH+ldH0NgfVLqofeFUg1j0bke1n+7DLf+Oh4ZiYZavciJjhuYx561TkkSEd1gqQGImEfkmKmhz82E8curx6uJW+kD9WagNbLj4FvEsBETIM7BSE8sei2vOVO42OnzvCej+P98IUelEqBS6sTeUNp79f8cC8n7nSYUJwTpshturBFoIFfcVQNoebdaao6gWpnxhUEnsc36AMl0DpIT0DsA9SxZQRjqhAEZmAWmASJMc5t/jWIjmJrqQxDy80HGJPEhIMTOAZfAiFqwyQwYTvJ0tOTW+jk2P9oriLHwi0Pr88uPGD5HWS6ULPoAutQAWMKCYgb5PoOiB8vbUZT9RTXfdLkHLkpw0ICaj5sam2o32CMLm8KZzfNVLJ/kUgdnkYUpC0vo+PGfguiJX8fvZxB06+niljW9iHteqVuoP1tADQwby4jS2ob3eL7TgCGuGk6gQkvHkLmOzlagKeXu1YZ0ow5vq6C+vSEfKEzIDrhXk="
 
+services:
+  - docker
+
 # todo: remove this once travis fixes their upstream issue (travis-ci/travis-ci#8836)
 sudo: required
-addons:
-  chrome: stable
 
-script:
-  # Bazel build & test everything. Can't use `bazel build ...` yet, because
-  # javaharness doesn't build on Travis (needs working Android SDK).
-  - >
-    bazel build --noshow_progress --noshow_loading_progress
-    //particles/...
-    //src/...
-    //src_kt/...
-  - >
-    bazel test --noshow_progress --noshow_loading_progress --test_output=errors
-    //particles/...
-    //src/...
-    //src_kt/...
-  - npm run test-with-start
+# Bazel build & test everything. Can't use `bazel build ...` yet, because
+# javaharness doesn't build on Travis (needs working Android SDK).
+jobs:
+  include:
+    - stage: build
+      script: tools/run_in_docker "bazel --bazelrc=tools/travis.bazelrc build //src_kt/..."
+    - stage: test
+      script: tools/run_in_docker "bazel --bazelrc=tools/travis.bazelrc test //src_kt/..."
+      name: "Bazel Test: //src_kt/..."
+    - script: tools/run_in_docker  "bazel --bazelrc=tools/travis.bazelrc test //src/..."
+      name: "Bazel Test: //src/..."
+    - script: tools/run_in_docker "bazel --bazelrc=tools/travis.bazelrc test //particles/..."
+      name: "Bazel Test: //particles/..."
+    - script: tools/run_in_docker "npm run test-no-web"
+      name: "Node Test: Unit"
+#     # TODO(alxr): Add back into CI once docker issue fixed
+#    - script: tools/run_in_docker "npm run test-shells"
+#      name: "Node Test: Shells"
 
+# Use travis to cache a Bazel disk-cache.
+# This connects to the docker container via a volume mount (`-v /home/travis/disk-cache:/disk-cache/`).
+# Bazel will store artifacts here due to the `--disk_cache=/disk-cache/` argument.
 cache:
   npm: true
+  directories:
+    - /home/travis/build/PolymerLabs/arcs/disk-cache/
```

Comment:
Didn't we decide that caching bazel artifacts didn't work properly? Do we still want to do this?

Reply:
Looking at how much extra this disk-cache adds, I'd suggest removing it for now so that we can more confidently land this PR. We can add it back in if it's helpful afterwards.

Reply:
Sure, we'll wait for verification. 

I think if we get a cache pruning strategy, this would again be useful.